### DISCUSSION TITLE
Sync Plugins categorization with Obsidian Community Hub

### DISCUSSION THREAD

**Discussion by ramisedhom** [ramisedhom/obsidian-plugins-categorization#15] [Ideas]
- Bidirectional synchronization between effort here and community contributions on ﻿[Obsidian Community Hub﻿](https://publish.obsidian.md/hub).
- On Obsidian Community Hub, plugins categorized on page: [🗂️ 02.01 Plugins by Category](https://publish.obsidian.md/hub/02+-+Community+Expansions/02.01+Plugins+by+Category/%F0%9F%97%82%EF%B8%8F+02.01+Plugins+by+Category#%F0%9F%97%82%EF%B8%8F+02.01+Plugins+by+Category)
- There is a dependency on Export plugins per category to markdown #11 idea

**Comment by claremacrae** [comment]
I would be happy to explore working together with you on this. We are long way behind on categorising plugins in Tasks.

**Reply by ramisedhom** [reply]
It'll be my pleasure. Let me know what you are thinking of on ways of collaborating on this idea.

**Comment by ramisedhom** [comment]
I think first we need to process uncategorized plugins as much as possible. I'm trying to do so, today we have only 200 plugins uncategorized in [Seatable](https://link.ramisedhom.com/Obsidian-Plugins-Categorization)

Second, I would introduce some categories from [Community Hub](https://publish.obsidian.md/hub/02+-+Community+Expansions/02.01+Plugins+by+Category/%F0%9F%97%82%EF%B8%8F+02.01+Plugins+by+Category#%F0%9F%97%82%EF%B8%8F+02.01+Plugins+by+Category) into [Seatable](https://link.ramisedhom.com/Obsidian-Plugins-Categorization) to ensure synchronization after that is working smoothly.

**Comment by ramisedhom** [comment]
I had finished categorizing all plugins in Seatable 🎉🎉🎉

**Reply by ichmoimeyo** [reply]
Fantastic job - thanks Rami!

I can't believe you got it done so fast.

I found 5 dupes you may want to get rid of (red marks on the left side) ...

https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=T0C7

Suggestion: In the All Plugins view

https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=0000

... you may want to have a view listing only available plugins so that the total number syncs with that from obidian-Releases and Obsidian Plugin Stats.

https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=CK7m

Thanks again for all your hard work in getting the categorization done :)

**Reply by ichmoimeyo** [reply]
I built myself another table called Status that is linked to from a column also called Status that I added to my version of Plugins.
It functions the same way as the Categories table with its 1st column denoting the plugins status in my vault e.g. enabled-always, enabled-toggle, try--toggle, try-next, try-when, try-u(ninstalled), etc.
Here is a read-only version that I'm about to share for a week:
https://cloud.seatable.io/dtable/external-links/custom/P-Obsidian-ichmoimeyo-Plugins/?tid=h8Hs&vid=Hv83

**Reply by ramisedhom** [reply]
@ichmoimeyo Thanks for your help in finding duplicates. I'll need to debug the script to ensure no duplicates are created in the future.

For the only available plugins view, I agree with you. We should keep this view, just renamed it to: "Only available plugins".

**Comment by ramisedhom** [comment]
Here is the difference between categories in Community Hub and Seatable:

Legend:
- `=` Same in both
- `+` to be added
- `*` Modified a bit

| Group                       | Diff | Category Name in Community Hub                | Category Name in Seatable        |
| --------------------------- | ---- | --------------------------------------------- | -------------------------------- |
| Language support            |      |                                               |                                  |
|                             | =    | Chinese Language Plugins                      | Chinese                          |
|                             | =    | Japanese Language Plugins                     | Japanese                         |
|                             | +    | Plugins for Right-To-Left Languages           |                                  |
|                             | *    | Dictionary and Spellchecking Plugins          | Translate and Spellchecking      |
| UX plugins                  |      |                                               |                                  |
|                             | =    | Status bar plugins                            | Status bar                       |
|                             | +    | Theme settings plugins                        |                                  |
|                             | +    | Plugins Designed for Mobile                   | Mobile                           |
|                             | *    | Style and Appearance Plugins                  | Style                            |
|                             | +    | Side bar plugins                              |                                  |
|                             | +    | Pane Management plugins                       |                                  |
|                             | *    | Plugins that add or manage hotkeys            | Shortcut                         |
|                             | +    | Toolbar plugins                               |                                  |
| Navigation & search plugins |      |                                               |                                  |
|                             | =    | Plugins to navigate notes                     | Navigation                       |
|                             | =    | Search and query plugins                      | Search                           |
|                             | =    | Plugins to Enhance Workspaces                 | Workspace                        |
|                             | *    | Folder management plugins                     | File and Folder                  |
|                             | +    | Tag management plugins                        |                                  |
| File Management plugins     |      |                                               |                                  |
|                             | *    | Plugins to manage files and attachments       | File and Folder                  |
|                             | =    | Plugins for Video and Audio files             | Audio and Video                  |
|                             | *    | Plugins to convert content into Markdown      | Convert                          |
|                             | *    | Plugins to export Markdown content            | Convert                          |
| Note contents               |      |                                               |                                  |
|                             | =    | Plugins to manage or annotate PDF files       | PDF                              |
|                             | =    | Plugins to handle images                      | Image                            |
|                             | =    | Mindmapping plugins                           | Mindmap                          |
|                             | =    | Plugins to create charts                      | Chart                            |
|                             | +    | Plugins for Diagrams                          |                                  |
|                             | =    | Graph plugins                                 | Graph                            |
|                             | =    | Plugins with custom codeblock syntax          | Render                           |
|                             | *    | Plugins with custom views                     | View                             |
| Editor plugins              |      |                                               |                                  |
|                             | *    | Plugins for Editing Notes                     | Edit                             |
|                             | =    | Plugins to edit other file types              | Editor                           |
|                             | =    | Refactoring and auto-formatting plugins       | Refactor                         |
|                             | +    | Markdown formatting plugins                   |                                  |
|                             | =    | Vim-related Plugins                           | Vim                              |
|                             | +    | Template plugins                              | Template                         |
| Integration                 |      |                                               |                                  |
|                             | =    | Third-Party Integration Plugins               | Integration                      |
|                             | =    | Automation plugins                            | Automation                       |
|                             | =    | Backup plugins                                | Backup                           |
|                             | *    | Plugins to convert content into Markdown      | Convert                          |
|                             | *    | Plugins to export Markdown content            | Convert                          |
|                             | =    | Date and calendar plugins                     | Date                             |
|                             | =    | Plugins for Security and Privacy              | Security                         |
|                             | =    | Task management plugins                       | Task                             |
|                             | =    | Plugins to manage internal and external links | Link                             |
| Workflows                   |      |                                               |                                  |
|                             | +    | Plugins for Writers                           |                                  |
|                             | *    | Plugins for daily notes                       | Journal                          |
|                             | +    | Spaced Repetition Plugins                     | Spaced Repetition and Flashcards |
|                             | +    | Time Management Plugins                       |                                  |
| Miscellaneous               |      |                                               |                                  |
|                             | +    | Plugins for Chess                             | Game                             |
|                             | +    | Plugins for TTRPG                             | Game                             |
|                             | +    | Paid and subscription-based plugins           |                                  |
|                             | =    | Finance plugins                               | Financial                        |
|                             | *    | Mathjax and LaTeX Plugins                     | Math                             |
|                             |      |                                               | Latex                            |
|                             | =    | Uncategorized plugins                         |                                  |
| Not published               |      |                                               |                                  |
|                             | +    | Plugins in Beta                               |                                  |
|                             | +    | Unlisted plugins                              |                                  |
|                             | *    | Discontinued plugins                          |                                  |
| MOC                         |      |                                               |                                  |
|                             | +    | Desktop-only plugins                          |                                  |
|                             | +    | Mobile-compatible plugins                     |                                  |
|                             | +    | OCR Plugins                                   |                                  |
|                             | +    | Plugins for collaboration                     |                                  |
|                             | =    | Plugins for vault statistics                  | Statistic                        |
|                             | +    |                                               | Artificial Intelligence          |
|                             | +    |                                               | Canvas                           |
|                             | +    |                                               | Frontmatter                      |
|                             | +    |                                               | History                          |
|                             | +    |                                               | Miscellaneous                    |
|                             | +    |                                               | Plugin Manager                   |
|                             | +    |                                               | Development                      |
|                             | +    |                                               | Tracker                          |

In total:
- Same categories are 26 out of 70 categories (37%)
- Modified categories are 14 out of 70 categories (20%)
- Categories to be added are 30 out of 70 categories (43%)

Google sheet: https://docs.google.com/spreadsheets/d/15V71H_lHMqBy2B46duF4pKsqEw25P_IBMTVHKyz1ao4/edit?usp=sharing

**Comment by ichmoimeyo** [comment]
@ramisedhom I found 1 duplicate:
id=notion-like-tables|CORRECT is -> DataLoom
https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=mhy0

Is there any particular reason you chose "name" as the primary identification column? 
Would using ID instead not automatically avoid duplication?

**Reply by claremacrae** [reply]
> Would using ID instead not automatically avoid duplication?

That's a good point: plugin authors can change plugin names, but the ID is the thing that stays the same for the lifetime of the plugin.

**Reply by claremacrae** [reply]
Like this one which has been through these names:

- `Notion-Like Tables` ->
- `Dashboards` ->
- `DataLoom `

... in 2 weeks.... and the ID is still `notion-like-tables`

https://github.com/trey-wallis/obsidian-dataloom/commits/master/manifest.json

**Reply by ramisedhom** [reply]
I think I'm using the plugin ID as the primary identifier, I'll double check the script and confirm

**Reply by ghost** [reply]
Hi, I am interested in working on plugin categorization as well. I am wondering is it possible to integrate it into current obsidian surface, so that users can easily search for plugins they need within the application.

**Reply by ramisedhom** [reply]
Welcome @FelixLai12 , I believe it's possible and this is exactly the idea #7 , but I'm not Obsidian plugin developer so I don't know how unfortunately. Are you a js/typescript developer?

**Reply by ghost** [reply]
I am not either. Are there lots of people having the same pain point? like can you see how many people access to the table you build?

**Comment by ichmoimeyo** [comment]
Dupes - ID: wikipedia-search
https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=Uyr1

**Reply by ichmoimeyo** [reply]
NO DUPES

**Comment by ichmoimeyo** [comment]
3 Dupes - ID: local-quotes x3
https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=Uyr1

https://github.com/sundevista/local-quotes x2
AND
https://github.com/decatetsu/local-quotes REDIRECTS TO https://github.com/sundevista/local-quotes

**Comment by ichmoimeyo** [comment]
11 DUPES - 5 IDs: bmo-chatbot | cycle-through-panes | local-quotes | obsidian-collapse-all-plugin | obsidian-importer
https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=Uyr1

ID: bmo-chatbot  https://github.com/longy2k/obsidian-bmo-chatbot x2 (Correct Descr. " Generate and brainstorm ..." )

ID: cycle-through-panes https://github.com/Vinzent03/tab-switcher CORRECT
ID: cycle-through-panes https://github.com/phibr0/cycle-through-panes REDIRECTS TO ABOVE

ID: local-quotes  https://github.com/sundevista/local-quotes x2
ID: local-quotes  https://github.com/decatetsu/local-quotes REDIRECTS TO ABOVE

ID: obsidian-collapse-all-plugin https://github.com/OfficerHalf/obsidian-collapse-all x2 (Correct Descr. "Extends collapse/expand ..." )

ID: obsidian-importer https://github.com/obsidianmd/obsidian-importer x2 (Correct Descr. "Import data from Notion, Evernote, ...")

**Reply by ichmoimeyo** [reply]
https://cloud.seatable.io/workspace/38222/dtable/P-Obsidian%20Plugins%20Categorization/?tid=28r6&vid=Uyr1

x8 were Modified by "NULL" 

8 of the entries do NOT list "Rami" as the modifier.

who or what is making these entries?

**Comment by ramisedhom** [comment]
@ichmoimeyo : The daily script is the modifier of those rows with empty "Modified by"
